base/utf16.jl

```diff
@@ -110,7 +110,9 @@ function is_valid_utf16(data::AbstractArray{UInt16})
     return i > n || !utf16_is_surrogate(data[i])
 end
 
-is_valid_utf16(s::UTF16String) = is_valid_utf16(s.data)
+isvalid(str::UTF16String) = isvalid(UTF16String, str.data)
+is_valid_utf16(str::UTF16String) = isvalid(UTF16String, str.data)
+is_valid_utf16(data::AbstractArray{UInt16}) = isvalid(UTF16String, data)
```

Comment:
Yup, exactly.

(We really need to have more folks who are qualified to review changes to the unicode/strings codebase.  I'm not that person, but I can occasionally make quick glances through change sets like this.)